.github/workflows/integration_tests.yml

```diff
@@ -21,60 +25,103 @@ jobs:
+              echo "LINODE_TOKEN=${{ secrets.LINODE_TOKEN_USER_2 }}" >> $GITHUB_ENV
+              ;;
+            "USER_3")
+              echo "TEST_TAGS=nb,nbconfig,nbconfigs,nbnode,nbs,networkingip,obj,objbucket,objcluster,objkey,profile,rdns" >> $GITHUB_ENV
+              echo "LINODE_TOKEN=${{ secrets.LINODE_TOKEN_USER_3 }}" >> $GITHUB_ENV
+              ;;
+            "USER_4")
+              echo "TEST_TAGS=lkenodepool,lkeversions,region,regions,sshkey,sshkeys,stackscript,stackscripts,token,user,users,vlan,volume,volumes,vpc,vpcs" >> $GITHUB_ENV
+              echo "LINODE_TOKEN=${{ secrets.LINODE_TOKEN_USER_4 }}" >> $GITHUB_ENV
+              ;;
+          esac
+
+      - name: Set report filename
+        id: set-report-filename
+        run: echo "REPORT_FILENAME=$(date +'%Y%m%d%H%M')_terraform_test_report_${{ matrix.user }}.xml" >> $GITHUB_ENV
+
       - name: Run Integration Tests
         run: |
-          timestamp=$(date +'%Y%m%d%H%M')
-          report_filename="${timestamp}_terraform_test_report.xml"
-          if ! make int-test | go-junit-report -set-exit-code > "$report_filename"; then
+          if ! make TEST_TAGS="${{ env.TEST_TAGS }}" int-test | go-junit-report -set-exit-code > $REPORT_FILENAME; then
             echo "EXIT_STATUS=1" >> $GITHUB_ENV
           fi
-          cat "$report_filename"
+          cat $REPORT_FILENAME
         env:
-          LINODE_TOKEN: ${{ secrets.DX_LINODE_TOKEN }}
+          LINODE_TOKEN: ${{ env.LINODE_TOKEN }}
 
-      - name: Add additional information to XML report
-        run: |
-          filename=$(ls | grep -E '^[0-9]{12}_terraform_test_report\.xml$')
-          python tod_scripts/add_to_xml_test_report.py \
-          --branch_name "${GITHUB_REF#refs/*/}" \
-          --gha_run_id "$GITHUB_RUN_ID" \
-          --gha_run_number "$GITHUB_RUN_NUMBER" \
-          --xmlfile "${filename}"
-
-      - name: Upload test results to bucket
-        run: |
-          report_filename=$(ls | grep -E '^[0-9]{12}_terraform_test_report\.xml$')
-          python3 tod_scripts/test_report_upload_script.py "${report_filename}"
-        env:
-          LINODE_CLI_OBJ_ACCESS_KEY: ${{ secrets.LINODE_CLI_OBJ_ACCESS_KEY }}
-          LINODE_CLI_OBJ_SECRET_KEY: ${{ secrets.LINODE_CLI_OBJ_SECRET_KEY }}
+      - name: Upload Test Report
+        uses: actions/upload-artifact@v4
+        with:
+          name: test-report
+          path: '*.xml'
 
       - name: Test Execution Status Handler
         run: |
           if [[ "$EXIT_STATUS" != 0 ]]; then
-            echo "Test execution contains failure(s)"
-            exit $EXIT_STATUS
+            echo "Test execution contains failure(s), check Run Integration Tests step above"
+            exit $EXIT_STATUS 
           else
-            echo "Tests passed!"
+            echo "Tests Passed!"
           fi
+
+
+  process-upload-report:
+    runs-on: ubuntu-latest
+    needs: [integration_tests]
+    if: always() # Run even if previous job fails
```

Comment:
This if statement will ensure this step will run even if previous executions failed